spec/controllers/verify/finance_controller_spec.rb

```diff
@@ -80,32 +80,182 @@
+        )
+      end
+    end
+  end
+
+  describe '#show' do
+    before do
+      stub_subject
+    end
+
+    context 'when the background job is not complete yet' do
+      render_views
+
+      it 'renders a spinner and has the page refresh' do
+        get :show
+
+        expect(response).to render_template('shared/refresh')
+
+        dom = Nokogiri::HTML(response.body)
+        expect(dom.css('meta[http-equiv="refresh"]')).to be_present
+      end
+    end
+
+    context 'when the background job has timed out' do
+      let(:expired_started_at) do
+        Time.zone.now.to_i - Figaro.env.async_job_refresh_max_wait_seconds.to_i
+      end
+
+      before do
+        controller.idv_session.async_result_started_at = expired_started_at
+        controller.idv_session.params = { ccn: '12345678' }
+      end
+
+      it 'displays an error' do
+        get :show
+
+        expect(response).to render_template :new
+        expect(flash[:warning]).to include(t('idv.modal.financials.timeout'))
+      end
+
+      it 'tracks the failure as a timeout' do
+        stub_analytics
+        allow(@analytics).to receive(:track_event)
+
+        get :show
+
+        result = {
+          success: false,
+          errors: { timed_out: ['Timed out waiting for vendor response'] },
+        }
+
+        expect(@analytics).to have_received(:track_event).with(
+          Analytics::IDV_FINANCE_CONFIRMATION_VENDOR, result
+        )
+      end
+    end
+
+    context 'when the background job has completed' do
+      let(:result_id) { SecureRandom.uuid }
+
+      before do
+        controller.idv_session.async_result_id = result_id
+        VendorValidatorResultStorage.new.store(result_id: result_id, result: result)
+      end
+
       context 'when CCN is confirmed' do
+        let(:result) { Idv::VendorResult.new(success: true) }
+
         it 'redirects to phone page' do
-          put :create, idv_finance_form: { finance_type: :ccn, ccn: '12345678' }
+          controller.idv_session.params = { ccn: '12345678' }
+
+          get :show
 
           expect(flash[:success]).to eq(t('idv.messages.personal_details_verified'))
           expect(response).to redirect_to verify_address_url
 
           expected_params = { ccn: '12345678' }
           expect(subject.idv_session.params).to eq expected_params
         end
+
+        it 'tracks the successful submission with no errors' do
+          stub_analytics
+          allow(@analytics).to receive(:track_event)
+
+          get :show
+
+          result = {
+            success: true,
+            errors: {},
+          }
+
+          expect(@analytics).to have_received(:track_event).with(
+            Analytics::IDV_FINANCE_CONFIRMATION_VENDOR, result
+          )
+        end
       end
 
       context 'when CCN is not confirmed' do
+        let(:result) do
+          Idv::VendorResult.new(
+            success: false,
+            errors: { ccn: ['The ccn could not be verified.'] }
+          )
+        end
```

Comment:
The old tests relied on being able to give bad values directly to the vendorvalidator, but now that we have split this into a separate step, we need to save the results in redis for each step